dev-libs/rapidjson/rapidjson-1.1.0.recipe

```diff
@@ -0,0 +1,66 @@
+SUMMARY="A fast JSON parser and generator for C++ with both SAX/DOM style API"
+DESCRIPTION="RapidJSON is a JSON parser and generator for C++. \
+It was inspired by RapidXml. RapidJSON is small but complete. \
+RapidJSON is fast. Its performance can be comparable to strlen(). \
+RapidJSON is self-contained and header-only. RapidJSON is memory-friendly. \
+RapidJSON is Unicode-friendly."
+HOMEPAGE="https://github.com/Tencent/rapidjson"
+COPYRIGHT="2015 THL A29 Limited, a Tencent company
+	2015 Milo Yip"
+LICENSE="MIT
+	JSON"
+REVISION="1"
+SOURCE_URI="https://github.com/Tencent/rapidjson/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="bf7ced29704a1e696fbccf2a2b4ea068e7774fa37f6d7dd4039d0787f8bed98e"
+SOURCE_FILENAME="rapidjson-$portVersion.tar.gz"
+PATCHES="rapidjson-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	rapidjson$secondaryArchSuffix = $portVersion
+	lib:rapidjson$secondaryArchSuffix = $portVersion compat >= $portVersion
```

Comment:
@alaviss, would you say the following suggestion is OK?

```diff
 ARCHITECTURES="!x86_gcc2 x86_64"
 SECONDARY_ARCHITECTURES="x86"
+
+portVersionCompat="$portVersion compat >= ${portVersion%.*}"
+
 PROVIDES="
 	rapidjson$secondaryArchSuffix = $portVersion
-	lib:rapidjson$secondaryArchSuffix = $portVersion compat >= $portVersion
+	lib:rapidjson$secondaryArchSuffix = $portVersionCompat
 	"
```
For the current release of rapidjson, the last line will be equivalent to:
```
 	lib:rapidjson$secondaryArchSuffix = 1.1.0 compat >= 1.1
```

Reply:
I don't think this port provides a library actually. As @Oyami-Srk said earlier, this is an [header-only library][0]. In that case `$portVersion` is more than enough.

[0]: https://github.com/haikuports/haikuports/pull/2107#discussion_r164770882

Reply:
OK, thanks.
So, I guess this is what we need to change:
```diff
-	lib:rapidjson$secondaryArchSuffix = $portVersion compat >= $portVersion
+	lib:rapidjson$secondaryArchSuffix = $portVersion
```
Is this correct?

Reply:
You should try to build this recipe to see if there's any library generated. If not, the line should be changed into

    devel:rapidjson = $portVersion

And have the `devel` package removed. [Xorg protocol ports][0] are examples for header-only library

[0]: https://github.com/haikuports/haikuports/tree/master/x11-proto